src/main/java/carpetaddonsnotfound/mixins/TallFlowerBlock_ReplaceFlowersMixin.java

```diff
@@ -26,13 +25,13 @@ public boolean canReplace(BlockState state, ItemPlacementContext context) {
       //from TallFlowerBlock.canReplace
       return false;
     }
-    //From AbstrackBlock.canReplace but without Material.isReplacable check.
+    //From AbstrackBlock.canReplace but without BlockState.isReplacable check.
     return context.getStack().isEmpty() || !context.getStack().isOf(this.asItem());
   }
 
   @Override
   public void onStateReplaced(BlockState state, World world, BlockPos pos, BlockState newState, boolean moved) {
-    if (state.get(TallFlowerBlock.HALF) == DoubleBlockHalf.LOWER && newState.getMaterial() != Material.AIR) {
+    if (state.get(TallFlowerBlock.HALF) == DoubleBlockHalf.LOWER && !newState.isAir()) {
```

Comment:
Same thing as in FlowerBlock mixin, make sure it acts the same.

Reply:
As said in previous comment, all rules were tested prior to PR being opened.